scripts/windows/setup/IotEdgeSecurityDaemon.ps1

```diff
@@ -849,23 +904,23 @@ function Get-AgentRegistry {
     return 'index.docker.io'
 }
 
-function Stop-IotEdgeContainers {
+function Remove-IotEdgeContainers {
     $dockerExe = Get-DockerExePath
 
     if (-not (Test-IsDockerRunning 6> $null)) {
         return
     }
 
-    $runningContainersString = Invoke-Native "$dockerExe ps --format ""{{.ID}}""" -Passthru
+    $runningContainersString = Invoke-Native "$dockerExe ps --all --format ""{{.ID}}""" -Passthru
     [string[]] $runningContainers = $runningContainersString -split {$_ -eq "`r" -or $_ -eq "`n"} | where {$_.Length -gt 0}
 
     foreach ($containerId in $runningContainers) {
-        $inspectString = Invoke-Native "$dockerExe inspect ""$containerId""" -Passthru | ConvertFrom-Json
+        $inspectString = Invoke-Native "$dockerExe inspect ""$containerId""" -Passthru
         $inspectResult = ($inspectString | ConvertFrom-Json)[0]
 
         if ($inspectResult.Config.Labels.'net.azure-devices.edge.owner' -eq 'Microsoft.Azure.Devices.Edge.Agent') {
-            Invoke-Native "$dockerExe stop ""$containerId"""
-            Write-Verbose "Stopped container $($inspectResult.Name)"
+            Invoke-Native "$dockerExe rm --force ""$containerId"""
```

Comment:
I'm not sure if we want to delete the containers when the customer upgrades their iotedged?

Reply:
We intentionally want to, to ensure that the new version gets pulled.

Reply:
As discussed, the label does get applied to user modules too, so this isn't correct. And this still leaves the container image behind anyway.